### PR TITLE
Document behaviour of unlink on directories.

### PR DESCRIPTION
**Pull request by sbriais**

### CODE REVIEW

File: otherlibs/unix/unix.mli [hunk 1]
```diff
@@ -464,7 +464,13 @@ module LargeFile :
 
 
 val unlink : string -> unit
-(** Removes the named file. *)
+(** Removes the named file.
+
+    If the named file is a directory, raises:
+      * [EPERM] on POSIX compliant system, or
```
Comment (dbuenzli) [inline otherlibs/unix/unix.mli]: You could use an unordered list here: 

```
{ul
{- [EPERM] on POSIX compliant system}
{- [EISDIR] on Linux >= 2.1.132}
{- [EACCESS] on Windows}}
```

**Review by dbuenzli — APPROVED** [review]
I generally think we shouldn't document the errors of Unix's function and rely on the posix docs but given the platform discreptance on that one, it seems warranted. What Seb says is consistent with what I have [observed](https://github.com/dbuenzli/bos/blob/fc01578a1e13b315a0b337afa2c8cc34e10098f3/src/bos_os_path.ml#L89-L96).

File: otherlibs/unix/unix.mli (hunk 1 above)
Comment (sbriais) [reply otherlibs/unix/unix.mli]: Thanks!

**Comment by tomjridge** [comment]
This is an alternative to POSIX...

https://github.com/sibylfs/sibylfs_src/blob/master/fs_spec/src/t_fs_spec.lem_cppo#L4001

**Comment by gasche** [comment]
@tomjridge I don't understand your message, could you elaborate? If I understand correctly (I didn't know about the SybilFS work, thanks, I will have a look), this is OCaml code that specifies the possible specification-compliant behaviors of filesystem code, and the relevant part is in [the `RN_dir _` case of `fsop_unlink_checks`](https://github.com/sibylfs/sibylfs_src/blob/f94d294/fs_spec/src/t_fs_spec.lem_cppo#L4037-L4039), which indicates that `EISDIR` may be raised under Linux and `EPERM` may always be raised. (I guess Windows unix-emulation behavior is out of the scope of this work). So this would confirm what @sbriais and @dbuenzli are saying above, and the proposed documentation change is correct, do I understand correctly?

**Comment by gasche** [comment]
I decided to go ahead and merge, thanks.

**Comment by tomjridge** [comment]
Sorry - somehow I missed @gasche 's question.

Yes, what you say is right. Hopefully we don't have to run so many tests of file system behaviour anymore - it should all be there in sibylfs. Windows is not there, but I think the docker people were looking at the compatibility layer (?) and running some tests, so perhaps in the future...

Thanks

**Pull request merged**